core/storage/trie/trie_batches.hpp

```diff
@@ -28,6 +28,9 @@ namespace kagome::storage::trie {
     virtual outcome::result<std::tuple<bool, uint32_t>> clearPrefix(
         const BufferView &prefix,
         std::optional<uint64_t> limit = std::nullopt) = 0;
+
+    virtual outcome::result<std::shared_ptr<TrieBatch>> createChildBatch(
+        common::Buffer path) = 0;
```

Comment:
Don't need this method.
`dynamic_cast` (like in `ChildStorageExtension::ext_default_child_storage_root_version_1`) is sufficient for `TrieStorageProviderImpl::getChildBatchAt`

Reply:
Well, you have a point, but dynamic_cast is a pretty poor solution genrally, so I prefer to avoid it where possible without too much effort. So I'm not sure about this place.